### PR TITLE
path limit changes

### PR DESCRIPTION
**Pull request by meggm**
# Description
This PR is used to set the MaxPathLimit parameter in the csi-powerscale driver

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/263 |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Tested by setting and unsetting the environment variable in the helm chart of csi-powerscale. There is no issue with the code
- [X] Tested by creating the path id equal to and more than the X_CSI_MAX_PATH_LIMIT variable. Results are as expected

### CODE REVIEW

File: helm/csi-isilon/values.yaml [hunk 1]
```diff
@@ -332,3 +332,8 @@ podmon:
   #    - "--mode=node"
   #    - "--leaderelection=false"
   #    - "--driver-config-params=/csi-isilon-config-params/driver-config-params.yaml"
+
+# Set the below parameter for setting the path size limit
+# Deafult value None
+# Examples: 128, 256 (Depends on the driver type and allowed value for path size limit)
+maxPathLen:
```
Comment (randeepsharma) [inline helm/csi-isilon/values.yaml:339]: In all other cases we have set a default value.
In the scenario this is left empty, what is the default in the code. You can use the same value here as default

**Review by randeepsharma — CHANGES_REQUESTED** [review]

File: helm/csi-isilon/values.yaml (hunk 1 above)
Comment (randeepsharma) [reply helm/csi-isilon/values.yaml:339]: The format of comment which specifies what this does, also needs to be changed. Please look at description of other variable in their comments.

Comment (prablr79) [reply helm/csi-isilon/values.yaml:339]: is this volume size or complete path size of the given volume ? or path or full address for the end LUN ?

Comment (meggm) [reply helm/csi-isilon/values.yaml]: It is the complete path size of the given volume

Comment (meggm) [reply helm/csi-isilon/values.yaml]: > In all other cases we have set a default value. In the scenario this is left empty, what is the default in the code. You can use the same value here as default

Set the default value

Comment (meggm) [reply helm/csi-isilon/values.yaml]: > The format of comment which specifies what this does, also needs to be changed. Please look at description of other variable in their comments.

This has been fixed

File: helm/csi-isilon/values.yaml [hunk 2]
```diff
@@ -332,3 +332,8 @@ podmon:
   #    - "--mode=node"
   #    - "--leaderelection=false"
   #    - "--driver-config-params=/csi-isilon-config-params/driver-config-params.yaml"
+
+# maxPathLen: this parameter is used for setting the maximum Path length for the given volume.
+# Deafult value: 128
```
Comment (randeepsharma) [inline helm/csi-isilon/values.yaml:337]: typo

**Review by prablr79 — APPROVED** [review]
lgtm

File: helm/csi-isilon/values.yaml (hunk 2 above)
Comment (randeepsharma) [reply helm/csi-isilon/values.yaml:337]: update the go.mod to pick the latest gocsi with this change

Comment (meggm) [reply helm/csi-isilon/values.yaml]: Done

**Review by randeepsharma — APPROVED** [review]
LGTM

**Review by prablr79 — APPROVED** [review]

**Pull request merged**